src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as string;
```

Comment:
Casting `item[key]` to a string assumes that the value of the property will always be a string. This might not be the case and could lead to unexpected behavior if the values are not strings. To improve type safety, consider adding a runtime check to ensure that the key's value is indeed a string, or refactor the function to handle different types of key values.